src/test/java/lambda/part3/exercise1/Exercise1.java

```diff
@@ -0,0 +1,80 @@
+package lambda.part3.exercise1;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@SuppressWarnings({"WeakerAccess", "unused"})
+public class Exercise1 {
+
+    @Test
+    public void mapEmployeesToLengthOfTheirFullNames() {
+        List<Employee> employees = getEmployees();
+        List<Integer> lengths = new ArrayList<>();
+
+        // функция извлечения информации о человеке из объекта сотрудника personExtractor: Employee -> Person
+        Function<Employee, Person> personExtractor = Employee::getPerson;
+        // функция извлечения полного имени из информации о человеке fullNameExtractor: Person -> String
+        Function<Person, String> fullNameExtractor = Person::getFullName;
+        // функция извлечения длины из строки stringLengthExtractor: String -> Integer
+        Function<String, Integer> stringLengthExtractor = String::length;
+        // функция извлечения длины полного имени из сотрудника fullNameLengthExtractor: Employee -> Integer
+        Function<Employee, Integer> fullNameLengthExtractor = (p) -> stringLengthExtractor.apply(fullNameExtractor.apply(personExtractor.apply(p)));
```

Comment:
Also, you can use the "andThen" method instead of the "apply" method. I'm sure you know it )